src/main/java/forest/detector/service/HtmlParser.java

```diff
@@ -0,0 +1,137 @@
+        Tract tract = new Tract();
+        String url = "https://lk.ukrforest.com" + tractLink;
+        Document document = Jsoup.connect(url).get();
+        Element tbody = document.select("tbody").get(1); // table
+        Elements row = tbody.select("tr"); // List of rows
+        String[][] table = elementsToArray(row);
+        int k = 1;
+        for (int i = 0; i < table[0].length; i++) {
+            tract.setTicketNumber(ticketNumber);
+            tract.setQuarter(table[0][i]);
+            tract.setDivision(table[1][i]);
+            tract.setRange(table[2][i]);
+            tract.setArea(Float.parseFloat(table[3][i]));
+            tract.setForestType(table[4][i]);
+            tract.setGeneralAllowedExtent(Float.parseFloat(table[5][i]));
+            tract.setAllowedExtent(Float.parseFloat(table[6][i]));
+            tract.setCuttingStatus(table[7][i]);
+            tract.setContributor(table[8][i]);
+            tract.setMapId(table[9][i]);
+            System.out.print("   " + k);
+            //System.out.println(" tract");
+            tractRepository.addTract(tract);
+            k++;
+        }
+    }
+
+    public static boolean isLastPage(Elements nextButtonClass) throws IOException {
+        return nextButtonClass.last().select("li").attr("class").equals("next disabled");
+    }
+
+
+    public static void pageCounter() throws IOException {
+        int pageNumber = 1;
+        boolean isCounted = false;
+        System.out.print("Counting pages: ");
+        while (!isCounted) {
+            String url = "https://lk.ukrforest.com/forest-tickets/index?TicketSearchPublic[region_id]=10&page=" + pageNumber;
+            Document document = Jsoup.connect(url).get();
+            Element ul = document.select("ul").get(1); // table
+            Elements next = ul.select("li"); // List of rows
+            System.out.print(pageNumber + " ");
+            if (next.last().select("li").attr("class").equals("next disabled")) {
+                System.out.println("--> " + pageNumber);
+                isCounted = true;
+            }
+            pageNumber += 1;
+        }
+    }
+
+    public static String[][] elementsToArray(Elements elements) throws IOException {
+        int columnSize = elements.get(0).select("td").size();
+        int rowSize = elements.size();
+        String[][] array = new String[columnSize][rowSize];
+        for (int i = 0; i < array[0].length; i++) {
+            for (int k = 0; k < array.length; k++) {
+                if (k < array.length - 1) {
+                    array[k][i] = (elements.get(i).select("td").get(k).text()).replaceAll("'", "''");
+                } else {
+                    array[k][i] = elements.get(i).select("td").get(k).select("a").get(0).attr("href");
+                }
+            }
+        }
+        return array;
+    }
+
+    public static String sqlDate(String date) throws ParseException {
+        Date sqlDate = new SimpleDateFormat("dd.MM.yyyy").parse(date);
```

Comment:
try to use new Java Time API. Here are some examples: https://github.com/searhiy/java8-datetime